tests/legacy/ApiTest.php

```diff
@@ -1330,6 +1354,7 @@ public function testApiDirectMessagesDestroyWithId()
 	 */
 	public function testApiDirectMessagesDestroyWithIdAndVerbose()
 	{
+		/*
```

Comment:
To make a request, Representative for all other tests:
- move all tests into its own test classes at the new structure
- copy a allready defined test and adapt it so you can pass the `$request[]` into the `run()` class

I can help you if you want, but it costs me 3 nights yet to check which test were replaced by which new class and what are the right testing parameters for it ^^
Maybe you can transform them as far as you can or have time and I'll do the rest

Reply:
I'll take a stab at it.

Reply:
I tried my hand at moving the tests to relevant classes, but I don't have a local database to test, and I keep stumbling on static calls that I'm not sure how to abstract. Do you have any pointers for this?

Reply:
I have to see if I can find some time.

Reply:
> I tried my hand at moving the tests to relevant classes, but I don't have a local database to test, and I keep stumbling on static calls that I'm not sure how to abstract. Do you have any pointers for this?

Without a local database it's nearly impossible to add new Module tests. There are too many and too different DB calls to mock them all. What I normally do:
- check the logic of the `Module` class itself (without all calls)
- Try to add at least one "straight forward path" test (=> expected output)
- Call it with the current DB fixture
- Look inside the missing DB calls and retrieve the inner structure what's missing and add the missing parts inside the dbfixture

With that way, normally I can add a test with ~2-3 test deifnition loops :-)